### ISSUE TITLE
User Story - Edson Fuentes

### ISSUE THREAD

**Issue by fuenteed** [adulbrich/osu-fsf#26]
User story statement: 
As a faculty and staff member, I want my steps or distance to be easily tracked so that my contribution to my teams is smoothly uploaded.

Background:

Why do we want it?
- We want it so that progress for events are seamless and smoothly uploaded to events to track progress. 

Who is this for?
- This feature applies to the faculty and staff participating in these events at OSU.

Do we have data to support it? 
- We do have the data to support it. We have access to API's to track distance traveled or steps taken.

Solution/Requirements: 
- Functional Requirements:
1. A user should be able to easily start a session to track distance or steps
2. The system should be able to calculate the steps or distances easily.
3. The system should show the user their progress in the current session 
4. A user should be able to stop the session anytime.
5. The system should be able to take the progress from the user's session and upload it to their event progress.
- Non-Functional Requirements:
1. The system should be able to handle long distances traveled.
2. The system should be able to upload the progress made by the user within 3 seconds.
3. The system should be available to track and progress 24 hours a day except during maintenance periods. 

Design resources :
- Figma 

Dependencies:
Are we waiting on other user stories or tasks to complete? 
- There are two user stories that this user story is dependent on. One is being able to create events, another is being able to view a report of the team.

Are others user stories or tasks waiting on this issue to complete? 

- None

List of tasks and subtasks:
- Task 1: Create a UI
Firstly, we need to have a smooth and clear UI for OSU FSF to navigate the application.

- Task 2: Implement a session starter
Using either some sort of API of third party service, we need to be able to track distances or steps of the user in their current session.

-Task 3: Upload to team or individual progress
Using database calculations, we need to be able to take the user's progress (potentially make some calculations on that progress to get it to the right metric) and smoothly upload It to that event's progress.  

Estimate
- We guarantee that each task will take no longer than a week. So if it takes the total duration for each task, then this should take around 3 weeks, a little less than a month. More realistically, it'll take about two weeks to get basic implementation and functionality.

Open Questions 
Acceptance Criteria

- Scenario: User wants to upload progress to a given event.
 -- Given: The user navigates to the session starter page
 --When: The user selects "start session"
 --And: The user makes some progress on the event via distance traveled/steps/etc..
 --Then: The system measures and calculates the progress made
 --Given: The user can see the progress from their current session.
 --When: The user uploads the progress to their event
 --Then: The system enables the user to view that progress

Definition of done (in addition to your software development process document, e.g., can only be marked as done once the integration with task X is tested)
-Produced code for presumed functionalities
-Assumptions of User Story met
-Project builds without errors
-Project deployed on the test environment identical to production platform
-Tests on devices/browsers listed in the project assumptions passed
-Feature is tested against acceptance criteria
-Feature ok-ed by Product Manager
-Refactoring completed
-Any configuration or build changes documented
-Documentation updated
-Peer Code Review performed

**Comment by fuenteed** [comment]
User story statement: 
As a faculty and staff member, I want my steps or distance to be easily tracked so that my contribution to my teams is smoothly uploaded.

Background:

Why do we want it?
- We want it so that progress for events are seamless and smoothly uploaded to events to track progress. 

Who is this for?
- This feature applies to the faculty and staff participating in these events at OSU.

Do we have data to support it? 
- We do have the data to support it. We have access to API's to track distance traveled or steps taken.

Solution/Requirements: 
- Functional Requirements:
1. A user should be able to easily start a session to track distance or steps
2. The system should be able to calculate the steps or distances easily.
3. The system should show the user their progress in the current session 
4. A user should be able to stop the session anytime.
5. The system should be able to take the progress from the user's session and upload it to their event progress.
- Non-Functional Requirements:
1. The system should be able to handle long distances traveled.
2. The system should be able to upload the progress made by the user within 3 seconds.
3. The system should be available to track and progress 24 hours a day except during maintenance periods. 


Design resources :
- Figma 


Dependencies:
Are we waiting on other user stories or tasks to complete? 
- There are two user stories that this user story is dependent on. One is being able to create events, another is being able to view a report of the team.

Are others user stories or tasks waiting on this issue to complete? 

- None

List of tasks and subtasks:
- Task 1: Create a UI
Firstly, we need to have a smooth and clear UI for OSU FSF to navigate the application.

- Task 2: Implement a session starter
Using either some sort of API of third party service, we need to be able to track distances or steps of the user in their current session.

-Task 3: Upload to team or individual progress
Using database calculations, we need to be able to take the user's progress (potentially make some calculations on that progress to get it to the right metric) and smoothly upload It to that event's progress.  

Estimate
- We guarantee that each task will take no longer than a week. So if it takes the total duration for each task, then this should take around 3 weeks, a little less than a month. More realistically, it'll take about two weeks to get basic implementation and functionality.

Open Questions 
Acceptance Criteria

- Scenario: User wants to upload progress to a given event.
 -- Given: The user navigates to the session starter page
 --When: The user selects "start session"
 --And: The user makes some progress on the event via distance traveled/steps/etc..
 --Then: The system measures and calculates the progress made
 --Given: The user can see the progress from their current session.
 --When: The user uploads the progress to their event
 --Then: The system enables the user to view that progress


Definition of done (in addition to your software development process document, e.g., can only be marked as done once the integration with task X is tested)
-Produced code for presumed functionalities
-Assumptions of User Story met
-Project builds without errors
-Project deployed on the test environment identical to production platform
-Tests on devices/browsers listed in the project assumptions passed
-Feature is tested against acceptance criteria
-Feature ok-ed by Product Manager
-Refactoring completed
-Any configuration or build changes documented
-Documentation updated
-Peer Code Review performed

**Comment by adulbrich** [comment]
Duplicate of #30 - please consolidate
